pkg/jobs/jobs_test.go

```diff
@@ -1093,42 +1099,47 @@ func TestRegistryLifecycle(t *testing.T) {
 	})
 
 	t.Run("dump traces on cancel", func(t *testing.T) {
+		skip.UnderStress(t, "job cancellation semantics is flaky under stress")
```

Comment:
If it's flaky due to retries, does it make sense to check that at least a certain number of trace files are there?